Mathlib/Algebra/Star/NonUnitalSubalgebra.lean

```diff
@@ -636,6 +636,10 @@ theorem adjoin_toNonUnitalSubalgebra (s : Set A) :
 theorem subset_adjoin (s : Set A) : s ⊆ adjoin R s :=
   (Set.subset_union_left s (star s)).trans <| NonUnitalAlgebra.subset_adjoin R
 
+@[set_like 20]
+lemma mem_adjoin_of_mem (s : Set A) {x : A} (hx : x ∈ s) : x ∈ adjoin R s :=
+  subset_adjoin R s hx
```

Comment:
I'm not super happy about a) these new lemmas, and b) the magic 20 that we have to remember for all of them.

Can we instead tag the `subset` lemma directly, and have it detect `⊆` and automatically set the number to 20?

Reply:
@eric-wieser 20 isn't a magic number. I think anything above the default penalty (which is 1) would probably be fine. I just made it 20 so there is some breathing room.

I tried tagging the `subset` lemma, but that doesn't trigger because it's the wrong statement (after expanding, it's `x ∈ s → x ∈ (↑(adjoin R s) : Set A)`, not `x ∈ s → x ∈ adjoin R s`), and so `aesop` doesn't want to apply it.

Reply:
Would a `SetLike.mem_of_subset : s ⊆ ↑S → x ∈ s →x ∈ S` lemma be enough for Aesop? If so, we'd only need one not many!

Reply:
That works, and I've implemented this and removed the other lemmas.